### PR TITLE
new(driver,userspace/libsinsp,test): properly manage `fd` in connect exit event too

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area driver-kmod
/area driver-bpf
/area driver-modern-bpf
/area libsinsp
/area tests

**Does this PR require a change in the driver versions?**

Yep but it was already bumped during this release cycle.

/version driver-SCHEMA-version-minor

**What this PR does / why we need it**:

This improves connect resiliency when connect enter event is dropped.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
new(driver,userspace): send fd in connect exit event too to improve resiliency in case connect enter gets dropped.
```

### CODE REVIEW

File: test/e2e/tests/test_event_generator/test_network_activity.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def test_network_activity(sinsp, run_containers: dict):
         },
         {
             "container.id": generator_id,
-            "evt.args": SinspField.regex_field(fr'^res=0 tuple={ipv4_regex}->10\.2\.3\.4:8192 $'),
+            "evt.args": SinspField.regex_field(fr'^res=0 tuple={ipv4_regex}->10\.2\.3\.4:8192 fd=3(<4>)$'),
```
Comment (FedeDP) [inline test/e2e/tests/test_event_generator/test_network_activity.py:46]: @Molter73 do you know what should i add here? :D 
`fd=3(<4>)` is exactly the same param that the previous event is receiving (the `connect` enter event)...

Comment (FedeDP) [reply test/e2e/tests/test_event_generator/test_network_activity.py:46]: Uh i see, i need to add the full tuple because sinsp adds it :)

**Review by incertum — APPROVED** [review]
/approve

File: userspace/libscap/engine/gvisor/parsers.cpp [hunk 2]
```diff
@@ -641,9 +641,10 @@ static parse_result parse_connect(const char *proto, size_t proto_size, scap_siz
 			return ret;
 		}
 
-		ret.status = scap_event_encode_params(scap_buf, &ret.size, scap_err, PPME_SOCKET_CONNECT_X, 2,
+		ret.status = scap_event_encode_params(scap_buf, &ret.size, scap_err, PPME_SOCKET_CONNECT_X, 3,
```
Comment (FedeDP) [inline userspace/libscap/engine/gvisor/parsers.cpp:644]: @LucaGuerra @loresuso PTAL :)

**Comment by FedeDP** [comment]
@incertum @LucaGuerra  @leogr PTAL! Thanks!

File: userspace/libsinsp/test/events_net.ut.cpp [hunk 3]
```diff
@@ -180,15 +180,14 @@ TEST_F(sinsp_with_test_input, net_ipv4_connect_with_intermediate_event)
 	std::vector<uint8_t> socktuple = test_utils::pack_socktuple(reinterpret_cast<sockaddr*>(&client), reinterpret_cast<sockaddr*>(&server));
 	/* This should never happen but could cause strange outcomes */
 	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SOCKET_SENDTO_E, 3, client_fd, (uint32_t)102, scap_const_sized_buffer{socktuple.data(), socktuple.size()});
-	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SOCKET_CONNECT_X, 2, return_value, scap_const_sized_buffer{socktuple.data(), socktuple.size()});
+	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SOCKET_CONNECT_X, 3, return_value, scap_const_sized_buffer{socktuple.data(), socktuple.size()}, client_fd);
```
[the remark this answers is not in the document]
Comment (FedeDP) [reply userspace/libsinsp/test/events_net.ut.cpp:183]: I'd just append the `missing last parameter` test to the newly added one, and keep all the other tests synced with master event table. WDYT?

**Review by LucaGuerra — APPROVED** [review]
Thanks! LGTM!

**Review by incertum — APPROVED** [review]
/approve

**Pull request merged**